manifests/init.pp

```diff
@@ -39,9 +39,16 @@
     purge_values => $purge_values
   }
 
-  service{ 'wuauserv':
-    ensure => running,
-    enable => true,
+  case $::kernelmajversion {
```

Comment:
I've used a kernel check because I tought that it was the most effective way to identify the documented(known) versions of Windows that are supporting the "Trigger start" feature.  For future versions if they do support the feature then you are right it will require another code modification.

If you can provide a better way to identify if the trigger start feature is supported in any Windows version, that would be great.


Bell
Richard Gareau
Systèmes haute sécurité – High Security Systems671 La Gauchetière O. Suite 1231, Montréal Q.C.  H3B 2M8
T: (514) 870-3602<tel:(514)%20870-3602>


-------- Message d'origine --------
De : "Ethan J. Brown"
Date :2016/07/11 17:37 (GMT-05:00)
À : puppetlabs/puppetlabs-wsus_client
Cc : "Gareau, Richard" , Mention
Objet : Re: [puppetlabs/puppetlabs-wsus_client] MODULES/2420-Make wuauserv service be idempotent (#50)


In manifests/init.pp<https://github.com/puppetlabs/puppetlabs-wsus_client/pull/50#discussion_r70342024>:

> @@ -39,9 +39,16 @@
>      purge_values => $purge_values
>    }
>
> -  service{ 'wuauserv':
> -    ensure => running,
> -    enable => true,
> +  case $::kernelmajversion {


Is the kernel check is always the right thing to do? What about future kernels, etc?

FYI - on my 2008R2 (kernel 6.1) that's supposed to support trigger starts, I get this:

C:\Users\Administrator>puppet resource service wuauserv
service { 'wuauserv':
  ensure => 'running',
  enable => 'true',
}

C:\Users\Administrator>sc queryex wuauserv

SERVICE_NAME: wuauserv
        TYPE               : 20  WIN32_SHARE_PROCESS
        STATE              : 4  RUNNING
                                (STOPPABLE, NOT_PAUSABLE, ACCEPTS_PRESHUTDOWN)
        WIN32_EXIT_CODE    : 0  (0x0)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0
        PID                : 876
        FLAGS              :

C:\Users\Administrator>sc qtriggerinfo wuauserv
[SC] QueryServiceConfig2 SUCCESS

    The service wuauserv has not registered for any start or stop triggers.


Note that I haven't used the module to configure WSUS settings though.

On Windows 10, in contrast:

$ sc qtriggerinfo wuauserv
[SC] QueryServiceConfig2 SUCCESS

SERVICE_NAME: wuauserv

        START SERVICE
          GROUP POLICY                 : 659fcae6-5bdb-4da9-b1ff-ca2a178d46e0 [MACHINE POLICY PRESENT]
        START SERVICE
          GROUP POLICY                 : 54fb46c8-f089-464c-b1fd-59d1b62c3b50 [USER POLICY PRESENT]


Checking these values (or the equivalent registry settings) is probably a better guard than kernelversion.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/puppetlabs/puppetlabs-wsus_client/pull/50/files/e833230e1ad1d10253b71d98c712b2e1946be9a6#r70342024>, or mute the thread<https://github.com/notifications/unsubscribe/AHfTx2Xot5tGubBYgsUDFfMwMq0YZ_ZUks5qUreEgaJpZM4I74HX>.